specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -49,13 +48,298 @@ import type { WbMapping } from './mapping';
 import { parseWbMappings } from './mapping';
 import { fetchWbPickLists } from './pickLists';
 import { WbUploaded } from './Results';
-import { wbViewTemplate } from './Template';
-import { WbActions } from './WbActions';
-import { WbUtils } from './WbUtils';
+import { WbActionsComponent } from '../WbActions/WbActions';
+import { WbUtils, WbUtilsComponent } from './WbUtils';
 import { WbValidation } from './WbValidation';
+import { DataSetName } from './DataSetMeta';
+import { WbSpreadsheet } from './WbSpreadsheet';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { WbToolkit } from '../WbToolkit/WbToolkit';
 
 export type WbStatus = 'unupload' | 'upload' | 'validate';
 
+export type Workbench = {
+  dataset: Dataset;
+  cells: WbCellMeta;
+  disambiguation: Disambiguation;
+  validation: WbValidation;
+  data: RA<RA<string | null>>;
+  hot: Handsontable;
+  throttleRate: number;
+  mappings: WbMapping;
+  utils: WbUtils;
+  cellCounts: GetSet<WbCellCounts>;
+};
+
+export function WbViewReact({
+  dataset,
+  hotRef,
+  handleDatasetDelete,
+  triggerRefresh,
+  spreadsheetContainer,
+}: {
+  readonly dataset: Dataset;
+  readonly hotRef: React.RefObject<HotTable>;
+  readonly handleDatasetDelete: () => void;
+  readonly triggerRefresh: () => void;
+  readonly spreadsheetContainer: any;
+}): JSX.Element {
+  const data = React.useMemo<RA<RA<string | null>>>(
+    () =>
+      dataset.rows.length === 0
+        ? [Array.from(dataset.columns).fill(null)]
+        : dataset.rows,
+    [dataset.rows]
+  );
+
+  const isUploaded = React.useMemo<boolean>(
+    () => dataset.uploadresult !== null && dataset.uploadresult.success,
+    [dataset]
+  );
+
+  const mappings = React.useMemo(
+    (): WbMapping | undefined => parseWbMappings(dataset),
+    [dataset]
+  );
+
+  const throttleRate = Math.ceil(clamp(10, data.length / 10, 2000));
+
+  const [hasUnsavedChanges, spreadsheetChanged, spreadsheetUpToDate] =
+    useBooleanState();
+  const [hotIsReady, setHotToReady] = useBooleanState();
```

Comment:
instead of this state, could we get away with just having a `const [hot,setHot]` state, and only set that state when hot is ready?